source/fundamentals/gridfs.txt

```diff
@@ -0,0 +1,707 @@
+of the ``GridFSBucketOptions class`` class to 
+the ``GridFSBucket()`` constructor. The following table describes the properties in the
+``GridFSBucketOptions`` class:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Field 
+     - Description
+
+   * - ``BucketName`` 
+     - | The bucket name to use as a prefix for the files and chunks collections.
+         The default value is ``"fs"``.
+       | **Data type**: {+string-data-type+} 
+
+   * - ``ChunkSizeBytes`` 
+     - | The chunk size that GridFS splits files into. The default value is ``261120``.
+       | **Data type**: {+int-data-type+}
+
+   * - ``ReadConcern`` 
+     - | The read concern to use for bucket operations. The default value is the 
+         database's read concern.
+       | **Data type**: `ReadConcern <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.ReadConcern.html>`__
+
+   * - ``ReadPreference`` 
+     - | The read preference to use for bucket operations. The default value is the 
+         database's read preference.
+       | **Data type**: `ReadPreference <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.ReadPreference.html>`__
+
+   * - ``WriteConcern`` 
+     - | The write concern to use for bucket operations. The default value is the 
+         database's write concern.
+       | **Data type**: `WriteConcern <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.WriteConcern.html>`__
+
+The following example creates a bucket named ``"myCustomBucket"`` by passing an instance
+of the ``GridFSBucketOptions`` class to the ``GridFSBucket()`` constructor:
+
+
+.. literalinclude:: /includes/code-examples/GridFS.cs
+   :language: csharp
+   :dedent:
+   :start-after: // start-create-custom-bucket
+   :end-before: // end-create-custom-bucket
+
+.. _gridfs-upload-files:
+
+Upload Files
+------------
+
+You can upload files to a GridFS bucket by using the following methods:
+
+- ``OpenUploadStreamAsync()`` or ``OpenUploadStream()``: Opens a new upload stream to which
+  you can write file contents
+- ``UploadFromStreamAsync()`` or ``UploadFromStream()``: Uploads the contents of an existing
+  stream to a GridFS file
+
+The following sections describe these methods in more detail.
+
+.. _gridfs-open-upload-stream:
+
+Write to an Upload Stream
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use the ``OpenUploadStreamAsync()`` or ``OpenUploadStream()`` method to create an upload
+stream for a given file name. These methods accept the following parameters:
+
+.. list-table::
+
+   * - Parameter
+     - Description
+   * - ``filename``
+     - | The name of the file to upload.
+       | 
+       | **Data type**: {+string-data-type+}
+   * - ``options``
+     - | *Optional.* An instance of the ``GridFSUploadOptions`` class that specifies the
+         configuration for the upload stream. The default value is ``null``.
+       | 
+       | **Data type**: `GridFSUploadOptions <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.GridFS.GridFSBucket.GridFSUploadOptions.html>`__
```

Comment:
I; link is 404ing

Reply:
great catch. for some reason they didn't put the GridFSUploadOptions class in the GridFSBucket namespace